test/integration-tests/test-get-keyboard-interactivity-types.c

```diff
@@ -0,0 +1,32 @@
+/* This entire file is licensed under MIT
+ *
+ * Copyright 2020 William Wold
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "integration-test-common.h"
+
+static GtkWindow* window;
+
+static void callback_0()
+{
+    window = create_default_window();
+    gtk_layer_init_for_window(window);
+    guint version = gtk_layer_get_layer_shell_version();
+    ASSERT_EQ(gtk_layer_get_keyboard_interactivity_type(window), GTK_LAYER_SHELL_KEYBOARD_NONE, "%d");
+    gtk_layer_set_keyboard_interactivity_type(window, GTK_LAYER_SHELL_KEYBOARD_ON_DEMAND);
+    ASSERT_EQ(gtk_layer_get_keyboard_interactivity_type(window), version >= 4 ? GTK_LAYER_SHELL_KEYBOARD_ON_DEMAND : GTK_LAYER_SHELL_KEYBOARD_NONE, "%d");
+    gtk_widget_show_all(GTK_WIDGET(window));
+    ASSERT_EQ(gtk_layer_get_keyboard_interactivity_type(window), version >= 4 ? GTK_LAYER_SHELL_KEYBOARD_ON_DEMAND : GTK_LAYER_SHELL_KEYBOARD_NONE, "%d");
```

Comment:
Since this will always be run against the mock server, no need for the version checks. We can assume it's on the expected version.

To answer your question, yes, ideally we'd have additional tests for different protocols versions but we're not set up to easily implement that right now, so don't bother. I'll add those tests if/when I set up a way to make the mock server use different versions of protocols.